src/main/java/uk/gov/justice/framework/command/tools/CommandLineArgumentParser.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.justice.framework.command.tools;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.Logger;
+
+
+public class CommandLineArgumentParser {
+
+    private static final Logger logger = getLogger(CommandLineArgumentParser.class.getName());
+
+    private String[] args;
+    private Options options = new Options();
+
+
+    public CommandLineArgumentParser(String[] args) {
+
+        this.args = args;
+
+        options.addOption("h", "help", false, "show help.");
+        options.addOption("c", "command", true, "Framework command to execute, for example CATCHUP,SHUTTER, UNSHUTTER & REBUILD.");
+        options.addOption("ho", "host", true, "Host Remote or Localhost");
+        options.addOption("p", "port", true, "Wildfly management port");
+    }
+
+    public boolean parse() {
+        CommandLineParser parser = new BasicParser();
+
+        try {
+            CommandLine cmd = parser.parse(options, args);
+
+            if (cmd.hasOption("h")){
+                help();
+            }
+
+            if (isCommandNullOrEmpty(cmd) && isHostNullOrEmpty(cmd) && isPortNullOrEmpty(cmd)) {
```

Comment:
You don't need this if/else statement , you can "return isCommandNullOrEmpty(cmd) && isHostNullOrEmpty(cmd) && isPortNullOrEmpty(cmd)"

Reply:
We need to have number of parameters check as well